modules/oe_multilingual_selection_page/oe_multilingual_selection_page.install

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * @file
+ * Install, update and uninstall functions for the module.
+ */
+
+declare(strict_types = 1);
+
+use Drupal\administration_language_negotiation\Plugin\LanguageNegotiation\LanguageNegotiationAdministrationLanguage;
+use Drupal\language\Plugin\LanguageNegotiation\LanguageNegotiationUrl;
+use Drupal\language\Plugin\LanguageNegotiation\LanguageNegotiationSelected;
+use Drupal\language_selection_page\Plugin\LanguageNegotiation\LanguageNegotiationLanguageSelectionPage;
+
+/**
+ * Implements hook_install().
+ */
+function oe_multilingual_selection_page_install() {
+  // Configure selection page language negotiation method.
+  \Drupal::configFactory()
+    ->getEditable('language_selection_page.negotiation')
+    ->set('path', '/select-language')
+    ->set('type', 'standalone')
+    ->set('ignore_neutral', FALSE)
+    ->set('blacklisted_paths', [
+      '/admin',
+      '/user',
+      '/admin/*',
+      '/admin*',
+      '/node/add/*',
+      '/node/*/edit',
+    ])->save();
```

Comment:
This configuration is not provided by the `oe_multilingual_selection_page` so it cannot be stored in its `config/install` module, doing that will result in an error at installation time.

Reply:
`config/optional` then?

Reply:
It would have the same effect, the configuration is in the config/install of the module this module depends on, so that will be installed first, causing the same error.

Reply:
OK let's go with the install hook, but there is a risk involved with this. If the site builder enables this module, then decides not to use it and uninstalls it again, the change to the language negotiation will not be undone. I would not suggest to revert to the previous language negotiation strategy either, since there might be a very long time between the enabling and uninstallation of the module, and the site may have become dependent on the language negotiation provided by this module.

What I think we need is a disclaimer / opt-in checkbox when enabling the module, and also documentation in the README that explains to site builders that enabling this module will set the administrative language to English.

Is there anything we can revert safely on `hook_uninstall()`? Maybe we can remove the `language_selection_page.negotiation` part from config and the `LanguageNegotiationLanguageSelectionPage::METHOD_ID` from the language negotiation?

I don't thing we can touch the other things on uninstall though :-/

Providing a disclaimer and opt-in checkbox on installation is out of scope for this ticket, let's do this in a followup after discussion. We can put a `@todo` here though linking to the followup, and make a note in the README.